uni-stark/examples/prove_monty_31_keccak.rs

```diff
@@ -0,0 +1,326 @@
+            field_hash, compress,
+        );
+
+    let challenge_mmcs = ExtensionMmcs::<F, EF, _>::new(val_mmcs.clone());
+
+    let trace = match proof_goal {
+        "B" | "Blake3" | "BLAKE3" => {
+            let inputs = (0..num_hashes)
+                .map(|_| random::<[u32; 24]>())
+                .collect::<Vec<_>>();
+            p3_blake3_air::generate_trace_rows(inputs)
+        }
+        "P" | "Poseidon2" | "POSEIDON2" => {
+            let inputs = (0..num_hashes)
+                .map(|_| random::<[F; P2_WIDTH]>())
+                .collect::<Vec<_>>();
+            p2_air.generate_vectorized_trace_rows(inputs)
+        }
+        "K" | "Keccak" | "KECCAK" => {
+            let inputs = (0..num_hashes)
+                .map(|_| random::<[u64; 25]>())
+                .collect::<Vec<_>>();
+            p3_keccak_air::generate_trace_rows(inputs)
+        }
+        _ => {
+            panic!("Could not understand the proof goal. Use \"B\" to prove Blake3 Hashes or \"K\" for Keccak hashes.")
+        }
+    };
+
+    let fri_config = FriConfig {
+        log_blowup: 1,
+        num_queries: 100,
+        proof_of_work_bits: 16,
+        mmcs: challenge_mmcs,
+    };
+
+    let pcs = TwoAdicFriPcs::new(dft, val_mmcs, fri_config);
+
+    let config = StarkConfig::new(pcs);
+
+    let mut proof_challenger = SerializingChallenger32::from_hasher(vec![], byte_hash);
+    let mut verif_challenger = SerializingChallenger32::from_hasher(vec![], byte_hash);
+
+    match proof_goal {
+        "B" | "Blake3" | "BLAKE3" => {
+            let proof = prove(
+                &config,
+                &Blake3Air {},
+                &mut proof_challenger,
+                trace,
+                &vec![],
+            );
+
+            verify(
+                &config,
+                &Blake3Air {},
+                &mut verif_challenger,
+                &proof,
+                &vec![],
+            )
+        }
+        "P" | "Poseidon2" | "POSEIDON2" => {
+            let proof = prove(&config, &p2_air, &mut proof_challenger, trace, &vec![]);
+
+            verify(&config, &p2_air, &mut verif_challenger, &proof, &vec![])
+        }
+        "K" | "Keccak" | "KECCAK" => {
+            let proof = prove(
+                &config,
+                &KeccakAir {},
+                &mut proof_challenger,
+                trace,
+                &vec![],
+            );
+
+            verify(
+                &config,
+                &KeccakAir {},
+                &mut verif_challenger,
+                &proof,
+                &vec![],
+            )
+        }
+        _ => {
+            panic!("Could not understand the proof goal. Use \"B\" to prove Blake3 Hashes or \"K\" for Keccak hashes.")
+        }
+    }
+}
+
+fn report_result(result: Result<(), impl Debug>) {
+    if result.is_ok() {
+        println!("Proof Verified Successfully")
+    } else {
+        println!("{:?}", result.unwrap_err())
```

Comment:
nit: it might be best to panic, just to make it extra clear there was an unexpected error (e.g. my terminal will show a red ✘)